src/org/labkey/test/tests/query/QueryMetadataTest.java

```diff
@@ -0,0 +1,210 @@
+                new FieldDefinition("value", FieldDefinition.ColumnType.Integer),
+                new FieldDefinition("users", FieldDefinition.ColumnType.User),
+                new FieldDefinition("date", FieldDefinition.ColumnType.DateAndTime),
+                new FieldDefinition("samples", new FieldDefinition.IntLookup(getProjectName(),"exp.materials", TEST_SAMPLES)),
+                new FieldDefinition("selfLookup", new FieldDefinition.IntLookup(getProjectName(),"lists", TEST_LIST)));
+        var dgen = new IntListDefinition(TEST_LIST, "Key").setFields(listColumns)
+                .create(createDefaultConnection(), getProjectName());
+    }
+
+    @Test
+    public void testUpdateLookupFields()
+    {
+        var editPage = EditMetadataPage.beginAt(this, getProjectName(), "lists", TEST_LIST);
+        editPage.fieldsPanel()
+                .getField("selfLookup")
+                .setLabel("SelfLookup")
+                .setLookupValidatorEnabled(true)     // is a change, but does not appear in metadata view
+                .setLookup(new FieldDefinition.IntLookup(null, "lists", TEST_LIST)) // non-change
+                .setDescription("has new description");
+        editPage.fieldsPanel()
+                .getField("samples")
+                .setLabel("SamplesLookup")
+                .setLookup(new FieldDefinition.IntLookup(getProjectName(), "samples", TEST_SAMPLES))
+                .setLookupValidatorEnabled(true);   // is a change but does not appear in metadata view
+        editPage.clickSave();
+
+        String expectedXml = "<tables xmlns=\"http://labkey.org/data/xml\">\n" +
+                "  <table tableName=\"queryMetadataTestList\" tableDbType=\"NOT_IN_DB\">\n" +
+                "    <columns>\n" +
+                "      <column columnName=\"samples\">\n" +
+                "        <columnTitle>SamplesLookup</columnTitle>\n" +
+                "        <fk>\n" +
+                "          <fkDbSchema>samples</fkDbSchema>\n" +
+                "          <fkTable>queryMetadataSamples</fkTable>\n" +
+                "          <fkColumnName>RowId</fkColumnName>\n" +
+                "          <fkFolderPath>/QueryMetadataTest Project</fkFolderPath>\n" +
+                "        </fk>\n" +
+                "      </column>\n" +
+                "      <column columnName=\"selfLookup\">\n" +
+                "        <description>has new description</description>\n" +
+                "        <columnTitle>SelfLookup</columnTitle>\n" +
+                "      </column>\n" +
+                "    </columns>\n" +
+                "  </table>\n" +
+                "</tables>";
+        var queryPage = editPage.clickEditSource();
+        checker().verifyEquals("expect xml to show only the delta for value description",
+                expectedXml, queryPage.getMetadataXml());
+
+        // clean up after
+        EditMetadataPage.beginAt(this, getProjectName(), "lists", TEST_LIST).resetToDefault();
```

Comment:
Instead of only doing this at the end of a successful test case I recommend doing this in an `afterEach()` (if that exists in junit) to ensure the metadata is reset. Otherwise, if that is not available, then move these clean up statements to the beginning of the test cases so they defend themselves against others failing.